src/main/java/org/apache/pulsar/io/jcloud/sink/BlobStoreAbstractSink.java

```diff
@@ -144,31 +145,27 @@ public void close() throws Exception {
     public void write(Record<GenericRecord> record) throws Exception {
         final Long sequenceId = record.getRecordSequence().get();
         LOGGER.info("write message[recordSequence={}]", sequenceId);
-
-        String filepath = buildPartitionPath(record, partitioner, format);
-        ByteSource payload = bindValue(record, format);
-        Blob blob = blobStore.blobBuilder(filepath)
-                .payload(payload)
-                .contentLength(payload.size())
-                .build();
         int currentSize;
-
         rwlock.writeLock().lock();
         try {
-            incomingList.add(Pair.of(record, blob));
+            incomingList.add(record);
             currentSize = incomingList.size();
         } finally {
             rwlock.writeLock().unlock();
         }
         LOGGER.info("build blob success[recordSequence={}]", sequenceId);
         if (currentSize == sinkConfig.getBatchSize()) {
-            flushExecutor.submit(() -> flush());
+            flushExecutor.submit(this::flush);
```

Comment:
I noticed the default batch size is 10, which will result in write a parquet file for every 10 messages?

Reply:
Is such that.